code/go/0chain.net/miner/protocol_receive_integration_tests.go

```diff
@@ -308,6 +311,28 @@ func getNotarisationInfo(not *Notarization) *cases.NotarisationInfo {
 	}
 }
 
+// HandleNotarizedBlockMessage - handles a notarized block for a previous round.
+func (mc *Chain) HandleNotarizedBlockMessage(ctx context.Context,
+	msg *BlockMessage) {
+	mc.handleNotarizedBlockMessage(ctx, msg)
+}
+
+// handleVRFShare - handles the vrf share.
+func (mc *Chain) HandleVRFShare(ctx context.Context, msg *BlockMessage) {
+	state := crpc.Client().State()
+
+	if state.RoundHasFinalized != nil && state.RoundHasFinalized.Round+1 == int(msg.VRFShare.Round) && utils.IsSpamReceiver(state, msg.VRFShare.Round-1) {
```

Comment:
this round is finished already

Reply:
This is a VRF of the next round. What do you mean?